### ISSUE TITLE
Mysql TRUNCATE PARTITION syntax not supported in SQLFluff

### ISSUE THREAD

**Issue by noir19** [sqlfluff/sqlfluff#6723]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

When running SQLFluff lint on a SQL file containing Mysql's `ALTER TABLE ... TRUNCATE PARTITION` syntax, I encountered a parsing error.

My SQL file (`test.sql`):
```sql
ALTER TABLE :schema_nm.CUST TRUNCATE PARTITION PART:data_dt;
INSERT INTO  :schema_nm.CUST
(
    DATA_DT
    ,DATA_LOAD_TM
)
SELECT
    DATA_DT
    ,DATA_LOAD_TM
FROM :schema_nm.T_CUST
WHERE DATA_DT=':data_dt'
;
```

When running:
```bash
sqlfluff lint test.sql -d mysql
```

I get the following error:
```
== [test.sql] FAIL
L:   1 | P:   1 |  PRS | Line 1, Position 1: Found unparsable section: 'ALTER
                       | TABLE schema_nm.CUST TRUNCATE PART...'
WARNING: Parsing errors found and dialect is set to 'mysql'. Have you configured your dialect correctly?
All Finished 📜 🎉!
```

### Additional Information
When I remove the `ALTER TABLE` statement and only keep the `INSERT INTO` part, SQLFluff successfully processes the file and identifies formatting issues rather than parsing errors:

```
sqlfluff test.sql -d mysql
== [test.sql] FAIL
L:   3 | P:  12 | LT01 | Expected only single space before naked identifier.
                       | Found '  '. [layout.spacing]
L:   6 | P:   5 | LT04 | Found leading comma ','. Expected only trailing near
                       | line breaks. [layout.commas]
L:   6 | P:   6 | LT01 | Expected single whitespace between comma ',' and naked
                       | identifier. [layout.spacing]
L:   9 | P:   1 | LT02 | Expected indent of 4 spaces. [layout.indent]
L:  10 | P:   5 | LT04 | Found leading comma ','. Expected only trailing near
                       | line breaks. [layout.commas]
L:  10 | P:   6 | LT01 | Expected single whitespace between comma ',' and naked
                       | identifier. [layout.spacing]
L:  12 | P:  14 | LT01 | Expected single whitespace between naked identifier and
                       | raw comparison operator '='. [layout.spacing]
L:  12 | P:  25 | CV06 | Statements must end with a semi-colon.
                       | [convention.terminator]
All Finished 📜 🎉!
```

### Expected Behaviour

SQLFluff should be able to parse Mysql's `ALTER TABLE ... TRUNCATE PARTITION` syntax without errors.

### Observed Behaviour

none

### How to reproduce

Create a file test.sql with the SQL code above
Run sqlfluff lint test.sql -d mysql 

### Dialect

mysql

### Version

sqlfluff 3.3.0

### Configuration

[sqlfluff]
large_file_skip_byte_limit = 0
templater = placeholder
exclude_rules = ST05, AL01, RF01
max_line_length = 120

[sqlfluff:templater:placeholder]
param_style = colon

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by noir19** [comment]
I hope to make a correction
When I remove the `ALTER TABLE` statement and only keep the `INSERT INTO` part, SQLFluff successfully processes the file and identifies formatting issues rather than parsing errors:

```
sqlfluff lint src\sql\ob\tmp\test.sql -d mysql
== [src\sql\ob\tmp\test.sql] FAIL
L:   3 | P:  12 | LT01 | Expected only single space before naked identifier.
                       | Found '  '. [layout.spacing]
L:   6 | P:   5 | LT04 | Found leading comma ','. Expected only trailing near
                       | line breaks. [layout.commas]
L:   6 | P:   6 | LT01 | Expected single whitespace between comma ',' and naked
                       | identifier. [layout.spacing]
L:   9 | P:   1 | LT02 | Expected indent of 4 spaces. [layout.indent]
L:  10 | P:   5 | LT04 | Found leading comma ','. Expected only trailing near
                       | line breaks. [layout.commas]
L:  10 | P:   6 | LT01 | Expected single whitespace between comma ',' and naked
                       | identifier. [layout.spacing]
L:  12 | P:  14 | LT01 | Expected single whitespace between naked identifier and
                       | raw comparison operator '='. [layout.spacing]
L:  12 | P:  25 | CV06 | Statements must end with a semi-colon.
                       | [convention.terminator]
All Finished 📜 🎉!
```
